tools/logviewer/main.js

```diff
@@ -0,0 +1,241 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Logger = {
+  _entries: [],
+  _modules: [],
+  _components: [],
+
+  async initialize(e) {
+    if (e.target.files.length === 0) return;
+
+    const file = e.target.files[0];
+    const content = await file.text();
+    for (let line of content.split('\n')) {
+      this.parseLine(line);
+    }
+
+    this.createDateRange();
+    this.createModuleList();
+    this.createComponentList();
+    this.populateLogTable();
+  },
+
+  parseLine(line) {
```

Comment:
I'm noticing a lot of `return` statements in here while attempting to parse out the category and class. However, and we do often see a few log messages that don't come from our own `Logger` class and are instead generated from Qt's logging infrastructure instead. 

I'm left a little bit worried that we might be loosing some of those other log messages in this view by discarding anything that doesn't match the Category and Class syntax.

Reply:
Good point. Follow up to support qt logs and the wireguard ones